pkg/kn/core/root_test.go

```diff
@@ -40,6 +41,51 @@ func TestNewDefaultKnCommand(t *testing.T) {
 	})
 }
 
+func TestNewDefKnCmdWithArgsforCmdValiditySuccess(t *testing.T) {
+
+	pluginHandler := plugin.NewDefaultPluginHandler(plugin.ValidPluginFilenamePrefixes,
+		"/random/plugins", false)
+	args := []string{"kn", "service", "update", "--tag=0.13"}
+	rootCmd := NewDefaultKnCommandWithArgs(NewKnCommand(), pluginHandler, args, os.Stdin, os.Stdout, os.Stderr)
+	checkRootCmd(t, rootCmd)
+}
+
+func TestNewDefKnCmdWithArgsforPrimaryCmdValidityErr(t *testing.T) {
+
+	if os.Getenv("EXIT") == "1" {
```

Comment:
I am not sure I understand the need for this `os.Getenv(“EXIT”)`?

Can we test without it?

Reply:
This test runs a sub process. So initially we set the env variable and when the subprocess runs it check the variable to execute the block which terminates with os.Exit.